lib/features/pokemon/presentation/widgets/search_pokemon_widget.dart

```diff
@@ -33,9 +31,11 @@ class SearchPokemonWidget extends StatelessWidget {
             children: [
               ElevatedButton(
                 onPressed: () {
-                  selectedPokemonItem.changeNumber(
-                    newNumber: Random().nextInt(maxPokemonId),
-                  );
+                  ref.read(selectedPokemonItemProvider.notifier).changeNumber(
+                        newNumber: Random().nextInt(maxPokemonId),
+                      );
+                  final int newNumber = ref.read(selectedPokemonItemProvider) +1;
```

Comment:
Why line 37-38 is added? I think the function should run inside the "changeNumber" instead of in here. Is it safe to move it to the "changeNumber" function?

Reply:
I just realized that this 2 lines would search the pokemon using API directly without need to press "Search" button